source/javascripts/index.js

```diff
@@ -71,24 +75,63 @@ for (let index = 0; index < carouselList.length; index++) {
 
 // initiate custom calendar carousels
 const carouselCalendarList = document.querySelectorAll(
-  '[data-calendar-splide]'
+  "[data-calendar-splide]"
 );
 for (let index = 0; index < carouselCalendarList.length; index++) {
   const carousel = carouselCalendarList[index];
   const carouselInstances = [];
   carouselInstances[index] = new CarouselCalendar(carousel);
 }
 
-const chartWrap = document.getElementsByClassName('chartWrap');
+// setup yt videos
+const YT_SERVICE = "youtube.com";
+const loadYouTubeVideo = (videoId) => {
+  console.log("videoId", videoId);
+  const videoEl = document.getElementById(videoId);
+  const url = videoEl.dataset.url;
+  console.log("videoUrl", url);
+  const video = VideoPlayer.getOrCreateInstance(videoEl);
+  console.log("video", video);
+  video.setYouTubeVideo(url);
+};
+const manageVideo = (overlay) => {
+  const videoLoadButton = overlay.querySelector("button");
```

Comment:
There are a lot of `querySelector()` calls and direct DOM interactions which are not great in React based app.

Take a look at https://github.com/italia/designers.italia.it/blob/main/src/components/media-player/media-player.js where AFAIR everything is opaque behind the VideoPlayer object.

Also that implementation supports subtitles and probably has a few small gotchas sorted out.